tests/v2/validation/provisioning/rke2/delete_custom_cluster_test.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	password "github.com/rancher/rancher/tests/framework/extensions/users/passwordgenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type CustomClusterDeleteTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+	hardened           bool
+	kubernetesVersions []string
+	cnis               []string
+	nodeProviders      []string
+	psact              string
+	advancedOptions    provisioning.AdvancedOptions
+}
+
+func (c *CustomClusterDeleteTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *CustomClusterDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	c.kubernetesVersions = clustersConfig.RKE2KubernetesVersions
+	c.cnis = clustersConfig.CNIs
+	c.nodeProviders = clustersConfig.NodeProviders
+	c.hardened = clustersConfig.Hardened
+	c.psact = clustersConfig.PSACT
+	c.advancedOptions = clustersConfig.AdvancedOptions
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	c.kubernetesVersions, err = kubernetesversions.Default(c.client, clusters.RKE2ClusterType.String(), c.kubernetesVersions)
+	require.NoError(c.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(c.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(c.T(), err)
+
+	c.standardUserClient = standardUserClient
+}
+
+func (c *CustomClusterDeleteTestSuite) TestDeletingRKE2CustomCluster() {
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+	nodesAndRoles := clustersConfig.NodesAndRoles
+
+	if len(nodesAndRoles) == 0 {
+		c.T().Skip()
+	}
+
+	rolesPerNode := []string{}
+
+	for _, nodes := range nodesAndRoles {
+		var finalRoleCommand string
+		if nodes.ControlPlane {
+			finalRoleCommand += " --controlplane"
+		}
+		if nodes.Etcd {
+			finalRoleCommand += " --etcd"
+		}
+		if nodes.Worker {
+			finalRoleCommand += " --worker"
+		}
+		rolesPerNode = append(rolesPerNode, finalRoleCommand)
+	}
```

Comment:
Would this bit require a small refactor since Caleb's custom cluster pools refactor?